app/BUILD.bazel

```diff
@@ -128,72 +167,84 @@ VIEW_MODEL_FILES = [
-    "src/main/java/org/oppia/app/utility/OnClickableAreaClickedListener.kt",
-    "src/main/java/org/oppia/app/utility/RegionClickEvent.kt",
+# TODO(#1617): Remove genrules post-gradle
+# Files altered by genrules to be excluded in the app library
+ALTERED_VIEW_MODELS = [
+    "src/main/java/org/oppia/app/administratorcontrols/administratorcontrolsitemviewmodel/AdministratorControlsAccountActionsViewModel.kt",
+    "src/main/java/org/oppia/app/help/faq/FAQListViewModel.kt",
+    "src/main/java/org/oppia/app/help/HelpItemViewModel.kt",
+    "src/main/java/org/oppia/app/help/HelpListViewModel.kt",
+    "src/main/java/org/oppia/app/onboarding/OnboadingSlideViewModel.kt",
+    "src/main/java/org/oppia/app/onboarding/OnboardingViewModel.kt",
+    "src/main/java/org/oppia/app/parser/StringToFractionParser.kt",
+    "src/main/java/org/oppia/app/parser/StringToNumberParser.kt",
+    "src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt",
+    "src/main/java/org/oppia/app/player/state/itemviewmodel/ImageRegionSelectionInteractionViewModel.kt",
+    "src/main/java/org/oppia/app/topic/info/TopicInfoViewModel.kt",
 ]
 
-ANNOTATION_FILES = [
-    "src/main/java/org/oppia/app/activity/ActivityScope.kt",
-    "src/main/java/org/oppia/app/fragment/FragmentScope.kt",
-    "src/main/java/org/oppia/app/utility/KeyboardHelper.kt",
-]
+[genrule(
+    name = "update_" + file[0:-3],
+    srcs = [file],
+    outs = [file[0:-3] + "_updated.kt"],
+    cmd = '''
+    cat $(SRCS) |
+    sed 's/import org.oppia.app.R/import org.oppia.app.vm.R/g' > $(OUTS)
+    ''',
+)
+for file in ALTERED_VIEW_MODELS]
 
+# Source files and genrule targets for views library.
 VIEW_FILES = [
     "src/main/java/org/oppia/app/customview/interaction/FractionInputInteractionView.kt",
     "src/main/java/org/oppia/app/customview/interaction/NumericInputInteractionView.kt",
     "src/main/java/org/oppia/app/customview/interaction/TextInputInteractionView.kt",
-    "src/main/java/org/oppia/app/customview/LessonThumbnailImageView.kt",
-    "src/main/java/org/oppia/app/customview/SegmentedCircularProgressView.kt",
     "src/main/java/org/oppia/app/player/state/DragDropSortInteractionView.kt",
     "src/main/java/org/oppia/app/player/state/ImageRegionSelectionInteractionView.kt",
     "src/main/java/org/oppia/app/player/state/SelectionInteractionView.kt",
-    "src/main/java/org/oppia/app/profile/ProfileInputView.kt",
-    "src/main/java/org/oppia/app/utility/ClickableAreasImage.kt",
+    "src/main/java/org/oppia/app/shim/ViewBindingShim.kt",
+    "src/main/java/org/oppia/app/shim/ViewComponentFactory.kt",
     "src/main/java/org/oppia/app/view/ViewComponent.kt",
     "src/main/java/org/oppia/app/view/ViewScope.kt",
-    "src/main/java/org/oppia/app/shim/ViewBindingShimInterface.kt",
-    "src/main/java/org/oppia/app/shim/ViewComponentFactory.kt",
+    "update_src/main/java/org/oppia/app/customview/LessonThumbnailImageView",
```

Comment:
I wonder if it's possible to compute this list based on the ALTERED_VIEWS below to avoid potential disparity. Maybe something like:

```python
VIEW_FILES = [
    ...
] + [
    _get_genrule_target(altered_view)
    for altered_view in ALTERED_VIEWS
]

def _get_genrule_target(update_view):
    return "updated_%s" % updated_view[0:-3]
```

WDYT? Could also simplify other genrule scenarios, too.

Reply:
I think thats a really good idea! On it :)